### PR TITLE
Moves deployment list wipeout to the project-specific runtime instance

### LINKED ISSUE #1432: Deployment list gets wiped out if an IDE window is opened or closed
This is especially frustrating with long-running deployments like Flex, that disappear from the list and there is no indication of when they're finished, or the exit status. There is also no exception thrown or error logged anywhere. The Application Servers window only goes green for a bit. This is happening on Linux.

To reproduce, open or close an IDE window in another project with a populated deployment list. I only tried to open projects in the same window, although it might also be reproducible by opening in a new window.

### PR DESCRIPTION
**Pull request by nkibler**
Fixes #1432.

This works by moving the deployment list wipeout logic to the `AppEngineRuntimeInstance` and registering the project listener hooks to be project-specific.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java [hunk 1]
```diff
@@ -132,11 +139,18 @@ public void computeDeployments(@NotNull ComputeDeploymentsCallback callback) {
...
         runnable.cancel();
       }
       createdDeployments.clear();
     }
+
+    createdDeployments
+        .keys()
+        .forEach(
+            project ->
+                ProjectManager.getInstance()
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java]: Is this not handled by the platform as part of cleanup on project close?

Comment (chanseokoh) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java]: Even before that, `createdDeployments` is cleared just above. It's empty.

Comment (nkibler) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java]: Haha whoops, good call @chanseokoh. And @etanshaul, I'm not sure. Is there any way to check that? During deployment, is there something I can monitor (e.g. underlying process) that would close correctly if the platform was doing this?

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java]: look at `ProjectManagerImpl#addProjectManagerListener`

looks like its registering the removal callback with the disposer on each add

Comment (nkibler) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java]: Hmm I only see that when it's explicitly passed a `Disposable`:

```
@Override
  public void addProjectManagerListener(@NotNull final ProjectManagerListener listener, @NotNull Disposable parentDisposable) {
    addProjectManagerListener(listener);
    Disposer.register(parentDisposable, () -> removeProjectManagerListener(listener));
  }
```

**Pull request merged**